docs/guides/latest/auto-backup/database.md

```diff
@@ -11,3 +11,604 @@ product_name: stash
+
+**Create Storage Secret:**
+
+At first, let's create a Storage Secret for the GCS backend,
+
+```console
+$ echo -n 'changeit' > RESTIC_PASSWORD
+$ echo -n '<your-project-id>' > GOOGLE_PROJECT_ID
+$ mv downloaded-sa-json.key > GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+$ kubectl create secret generic -n demo gcs-secret \
+    --from-file=./RESTIC_PASSWORD \
+    --from-file=./GOOGLE_PROJECT_ID \
+    --from-file=./GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+secret/gcs-secret created
+```
+
+**Create BackupBlueprint:**
+
+Now, we have to create a `BackupBlueprint` crd with a blueprint for `Repository` and `BackupConfiguration` object.
+
+Below is the YAML of the `BackupBlueprint` object that we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupBlueprint
+metadata:
+  name: pg-backup-blueprint
+spec:
+  # ============== Template for Repository ==========================
+  backend:
+    gcs:
+      bucket: appscode-qa
+      prefix: stash-backup/${TARGET_NAMESPACE}/${TARGET_APP_GROUP}/${TARGET_APP_RESOURCE}/${TARGET_NAME}
+    storageSecretName: gcs-secret
+  # ============== Template for BackupConfiguration =================
+  task:
+    name: pg-backup-${TARGET_APP_VERSION}
+  schedule: "*/5 * * * *"
+  retentionPolicy:
+    name: "keep-last-5"
+    keepLast: 5
+    prune: true
+```
+
+Here,
+
+- `spec.task.name` specifies the `Task` crd name that will be used to backup the targeted database. We have used a variable `${TARGET_APP_VERSION}` as task name suffix. This variable will be substituted by the respective database version. This allows to backup multiple database versions with a common blueprint.
+
+Note that we have used some variables (format: `${<variable name>}`) in `spec.backend.gcs.prefix` field. Stash will substitute these variables with values from the respective target. To learn which variables you can use in the `prefix` field, please visit [here](/docs/concepts/crds/backupblueprint.md#repository-blueprint).
+
+Let's create the `BackupBlueprint` that we have shown above,
+
+```console
+$ kubectl apply -f ./docs/examples/guides/latest/auto-backup/database/backupblueprint.yaml
+backupblueprint.stash.appscode.com/pg-backup-blueprint created
+```
+
+Now, automatic backup is configured for PostgreSQL database. We just have to add a annotation to the `AppBinding` of the targeted database.
+
+> Note: `BackupBlueprint` is a non-namespaced crd. So, you can use a `BackupBlueprint` to backup targets in multiple namespaces. However, Storage Secret is a namespaced object. So, you have to manually create the secret in each namespace where you have a target for backup. Please give us your feedback on how to improve the ux of this aspect of Stash on [GitHub](https://github.com/stashed/stash/issues/842).
+
+**Required Annotation for Auto-Backup Database:**
+
+You have to add the following annotation to the `AppBinding` crd of the targeted database to enable backup for it:
```

Comment:
This label will not stick given how we generate labels for KubeDB appbindings https://github.com/kubedb/postgres/blob/master/pkg/controller/appbinding.go#L30

This should be applied to the Database CRD and passed to AppBinding.

cc: @the-redback . Let's talk tomorrow.

Reply:
So, we forward labels, not annotations. So, this should be fine. 

TODO: Stop setting annotations on AppBinding in KubeDB.

Reply:
done